components/net/cookie.rs

```diff
@@ -139,6 +149,9 @@ impl Cookie {
 
     // http://tools.ietf.org/html/rfc6265#section-5.1.3
     pub fn domain_match(string: &str, domain_string: &str) -> bool {
+        let string = &string.to_lowercase();
+        let domain_string = &domain_string.to_lowercase();
```

Comment:
You should do a case-insensitive comparison with the `unicase`, and I am not sure the case-insensitivity should apply to all Unicode code points, just ASCII. Do you know where the spec states such things should be case-insensitive?